src/App.tsx

```diff
@@ -1,26 +1,255 @@
+  };
+
+  const handleCloseErrorMessage = () => {
+    setErrorMessage('');
+  };
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const filterTodos = useCallback(
+    (currentTodos: Todo[], currentFilter: Filter) => {
+      if (currentFilter === Filter.Active) {
+        return currentTodos.filter((todo: Todo) => !todo.completed);
+      } else if (currentFilter === Filter.Completed) {
+        return currentTodos.filter((todo: Todo) => todo.completed);
+      } else {
+        return currentTodos;
+      }
+    },
+    [],
+  );
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const filteredTodos = useMemo(() => {
+    return filterTodos(todos, filter);
+  }, [filterTodos, todos, filter]);
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className={cn('todoapp__toggle-all', {
+              active: todos.every(todo => todo.completed),
+            })}
+            data-cy="ToggleAllButton"
+          />
+          {/* Add a todo on form submit */}
+          <form onSubmit={handleSubmit}>
+            <input
+              onChange={handleTitleChange}
+              ref={titleField}
+              value={title}
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              disabled={isSubmiting}
+            />
+          </form>
+        </header>
+
+        {!loading && (
+          <>
+            <TodoList
+              todos={filteredTodos}
+              handleToggleTodo={handleToggleTodo}
+              onDeleteTodo={handleDeleteTodo}
+              loadingTodoIds={loadingTodoIds}
+            />
+            {tempTodo !== null && (
+              <TodoItem todo={tempTodo} handleToggleTodo={handleToggleTodo} />
+            )}
+          </>
+        )}
+        {todos.length > 0 && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {active.length} items left
+            </span>
+
+            <TodoFilterPanel
+              filter={filter}
+              onFilterChange={handleFilterChange}
+            />
+            <button
+              type="button"
+              className="todoapp__clear-completed"
+              data-cy="ClearCompletedButton"
+              disabled={!isDisabled}
```

Comment:
Consider changing logic to use `isDisabled` without `!` because right now tour button disabled if your `isDisabled` value is false